From 02bc79bcd0eea1bd5d76c612f14586b8baa60517 Mon Sep 17 00:00:00 2001
From: Simon Fels <simon.fels@canonical.com>
Date: Fri, 22 Jul 2016 11:36:44 +0200
Subject: [PATCH 1/2] obexd: use system bus for communication

On snappy systems we don't nessarily have a session bus available so we
will run obexd on the system bus.
---
 obexd/client/ftp.c     | 2 +-
 obexd/client/map.c     | 2 +-
 obexd/client/opp.c     | 2 +-
 obexd/client/pbap.c    | 2 +-
 obexd/client/session.c | 2 +-
 obexd/client/sync.c    | 2 +-
 obexd/src/manager.c    | 2 +-
 tools/obexctl.c        | 2 +-
 8 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/obexd/client/ftp.c b/obexd/client/ftp.c
index 5e30654..3b82f3e 100644
--- a/obexd/client/ftp.c
+++ b/obexd/client/ftp.c
@@ -476,7 +476,7 @@ int ftp_init(void)
 
 	DBG("");
 
-	conn = dbus_bus_get(DBUS_BUS_SESSION, NULL);
+	conn = dbus_bus_get(DBUS_BUS_SYSTEM, NULL);
 	if (!conn)
 		return -EIO;
 
diff --git a/obexd/client/map.c b/obexd/client/map.c
index 23be2d6..28e60fe 100644
--- a/obexd/client/map.c
+++ b/obexd/client/map.c
@@ -2076,7 +2076,7 @@ int map_init(void)
 
 	DBG("");
 
-	conn = dbus_bus_get(DBUS_BUS_SESSION, NULL);
+	conn = dbus_bus_get(DBUS_BUS_SYSTEM, NULL);
 	if (!conn)
 		return -EIO;
 
diff --git a/obexd/client/opp.c b/obexd/client/opp.c
index 92785f6..36fd4db 100644
--- a/obexd/client/opp.c
+++ b/obexd/client/opp.c
@@ -191,7 +191,7 @@ int opp_init(void)
 
 	DBG("");
 
-	conn = dbus_bus_get(DBUS_BUS_SESSION, NULL);
+	conn = dbus_bus_get(DBUS_BUS_SYSTEM, NULL);
 	if (!conn)
 		return -EIO;
 
diff --git a/obexd/client/pbap.c b/obexd/client/pbap.c
index 3f5665f..e4b7d59 100644
--- a/obexd/client/pbap.c
+++ b/obexd/client/pbap.c
@@ -1314,7 +1314,7 @@ int pbap_init(void)
 
 	DBG("");
 
-	conn = dbus_bus_get(DBUS_BUS_SESSION, NULL);
+	conn = dbus_bus_get(DBUS_BUS_SYSTEM, NULL);
 	if (!conn)
 		return -EIO;
 
diff --git a/obexd/client/session.c b/obexd/client/session.c
index 5bd2d26..cf4cb2c 100644
--- a/obexd/client/session.c
+++ b/obexd/client/session.c
@@ -596,7 +596,7 @@ struct obc_session *obc_session_create(const char *source,
 	if (driver == NULL)
 		return NULL;
 
-	conn = dbus_bus_get(DBUS_BUS_SESSION, NULL);
+	conn = dbus_bus_get(DBUS_BUS_SYSTEM, NULL);
 	if (conn == NULL)
 		return NULL;
 
diff --git a/obexd/client/sync.c b/obexd/client/sync.c
index 548c318..8c30908 100644
--- a/obexd/client/sync.c
+++ b/obexd/client/sync.c
@@ -237,7 +237,7 @@ int sync_init(void)
 
 	DBG("");
 
-	conn = dbus_bus_get(DBUS_BUS_SESSION, NULL);
+	conn = dbus_bus_get(DBUS_BUS_SYSTEM, NULL);
 	if (!conn)
 		return -EIO;
 
diff --git a/obexd/src/manager.c b/obexd/src/manager.c
index c172c59..9b5bdde 100644
--- a/obexd/src/manager.c
+++ b/obexd/src/manager.c
@@ -499,7 +499,7 @@ gboolean manager_init(void)
 
 	dbus_error_init(&err);
 
-	connection = g_dbus_setup_bus(DBUS_BUS_SESSION, OBEXD_SERVICE, &err);
+	connection = g_dbus_setup_bus(DBUS_BUS_SYSTEM, OBEXD_SERVICE, &err);
 	if (connection == NULL) {
 		if (dbus_error_is_set(&err) == TRUE) {
 			fprintf(stderr, "%s\n", err.message);
diff --git a/tools/obexctl.c b/tools/obexctl.c
index 0709f69..9ed336a 100644
--- a/tools/obexctl.c
+++ b/tools/obexctl.c
@@ -2152,7 +2152,7 @@ int main(int argc, char *argv[])
 	bt_shell_set_menu(&main_menu);
 	bt_shell_set_prompt(PROMPT_OFF);
 
-	dbus_conn = g_dbus_setup_bus(DBUS_BUS_SESSION, NULL, NULL);
+	dbus_conn = g_dbus_setup_bus(DBUS_BUS_SYSTEM, NULL, NULL);
 
 	client = g_dbus_client_new(dbus_conn, "org.bluez.obex",
 							"/org/bluez/obex");
-- 
2.17.1

